servicetest/cgroups/test_cgroups.py

```diff
@@ -0,0 +1,178 @@
+# configurations to the Container helper. Tests that need to add, remove or
+# update entries can simply base their dict on this one for convenience.
+DATA = {
+    Container.CONFIG: '[{"enableWriteBuffer": false}]',
+    Container.BIND_MOUNT_DIR: "/gateways/app",
+    Container.HOST_PATH: CURRENT_DIR,
+    Container.READONLY: False
+}
+
+CONFIG_TEST_SMALL_THRESHOLD = [
+    {"setting": "memory.limit_in_bytes", "value": "1K"}
+]
+
+CONFIG_TEST_MEMORY_SHARE = [
+    {"setting": "memory.limit_in_bytes", "value": "1G"},
+    {"setting": "memory.memsw.limit_in_bytes", "value": "1G"}
+]
+
+CONFIG_TEST_WHITELISTING = [
+    {"setting": "memory.limit_in_bytes", "value": "200000"},
+    {"setting": "memory.limit_in_bytes", "value": "100"}
+]
+
+test_cap_0 = Capability("test.cap.small.threshold",
+                        [
+                            {"id": "cgroups", "config": CONFIG_TEST_SMALL_THRESHOLD}
+                        ])
+
+test_cap_1 = Capability("test.cap.memory.share",
+                        [
+                            {"id": "cgroups", "config": CONFIG_TEST_MEMORY_SHARE}
+                        ])
+
+test_cap_2 = Capability("test.cap.memory.whitelist",
+                        [
+                            {"id": "cgroups", "config": CONFIG_TEST_WHITELISTING}
+                        ])
+
+manifest = StandardManifest(TESTOUTPUT_DIR,
+                            "cgroup-test-manifest.json",
+                            [test_cap_0, test_cap_1, test_cap_2])
+
+
+def service_manifests():
+    """ The agent fixture calls this function when it creates the service manifests
+        that should be used with this test module. The agent fixture expects a list
+        of StandardManifest and/or DefaultManifest objects.
+    """
+    return [manifest]
+
+
+@pytest.fixture
+def create_testoutput_dir(scope="module"):
+    """ Create a directory for the generated test files.
+
+        This directory is ignored by git but it's nice to have
+        somewhere locally to store test output to support
+        troubleshooting etc.
+    """
+    if not os.path.exists(TESTOUTPUT_DIR):
+        os.makedirs(TESTOUTPUT_DIR)
+
+"""##### Test suites #####"""
+
+
+@pytest.mark.usefixtures("testhelper", "create_testoutput_dir", "agent")
+class TestCGroupGateway(object):
+    """ This suite tests that whether CGroupsGateway can working with supported options or not.
+
+        Prerequisites :
+        CONFIG_MEMCG and CONFIG_MEMCG_SWAP kernel options should have been enabled for these tests.
+        Also cgroup_enable=memory and swapaccount=1 options should be enabled by grub.
+    """
+    def test_small_threshold(self):
+        """ Test if the memory limiting with cgroup gateway is working as expected
+            which behavior is when allocated more memory than limit it should fail
+        """
+        try:
+            sc = Container()
+            sc.start(DATA)
+
+            success = sc.set_capabilities(["test.cap.small.threshold"])
```

Comment:
the return value of the function is not used in every case. But it is used this way when they are used. And also in many places in the testframework lib those kind of variables named as success .